src/plugins/data_views/common/data_views/data_view_lazy.ts

```diff
@@ -0,0 +1,619 @@
+              subField.type,
+              runtimeField,
+              name
+            );
+          });
+        } else {
+          addRuntimeFieldToSpecFields(name, runtimeField.type, runtimeField);
+        }
+      }
+    }
+
+    return spec;
+  };
+
+  private getScriptedFields({ fieldName = ['*'] }: Pick<GetFieldsParams, 'fieldName'>) {
+    const dataViewFields: Record<string, DataViewField> = {};
+
+    Object.values(this.scriptedFieldsMap).forEach((field) => {
+      if (!fieldMatchesFieldsRequested(field.name, fieldName)) {
+        return;
+      }
+      let fld = this.fieldCache.get(field.name);
+
+      if (fld) {
+        this.fieldCache.delete(field.name);
+      }
+      fld = new DataViewField({
+        ...field,
+        scripted: true,
+        searchable: false,
+        aggregatable: false,
+        count: this.fieldAttrs?.[field.name]?.count,
+        customLabel: this.fieldAttrs?.[field.name]?.customLabel,
+        customDescription: this.fieldAttrs?.[field.name]?.customDescription,
+      });
+      this.fieldCache.set(field.name, fld);
+      dataViewFields[field.name] = fld;
+    });
+    return dataViewFields;
+  }
+
+  private async getMappedFields({
+    fieldName,
+    forceRefresh = false,
+    unmapped: includeUnmapped,
+    indexFilter,
+    metaFields,
+    fieldTypes,
+  }: Omit<GetFieldsParams, 'mapped' | 'scripted' | 'runtime'>) {
+    const response = await this.apiClient.getFieldsForWildcard({
+      pattern: this.getIndexPattern(),
+      metaFields: metaFields ? this.metaFields : undefined,
+      type: this.type,
+      rollupIndex: this.typeMeta?.params?.rollup_index,
+      fields: fieldName || ['*'],
+      allowNoIndex: true,
+      indexFilter,
+      includeUnmapped,
+      forceRefresh,
+      fieldTypes,
+    });
+
+    const dataViewFields: Record<string, DataViewField> = {};
+    response.fields.forEach((field) => {
+      // keep existing field object, make sure content is fresh
+      let fld = this.fieldCache.get(field.name);
+      if (fld) {
+        // get fresh attributes
+        assign(fld.spec, field);
+        fld.spec.runtimeField = undefined; // unset if it was a runtime field but now mapped
+        fld.spec.isMapped = true;
+      } else {
+        fld = new DataViewField({
+          ...field,
+          count: this.fieldAttrs?.[field.name]?.count,
+          customLabel: this.fieldAttrs?.[field.name]?.customLabel,
+          customDescription: this.fieldAttrs?.[field.name]?.customDescription,
+          shortDotsEnable: this.shortDotsEnable,
+        });
+        this.fieldCache.set(field.name, fld);
+      }
+      dataViewFields[field.name] = fld;
+    });
+
+    return dataViewFields;
+  }
+
+  getScriptedFieldsForQuery() {
+    return Object.values(this.scriptedFieldsMap).reduce<Record<string, estypes.ScriptField>>(
+      (scriptFields, field) => {
+        scriptFields[field.name] = {
+          script: {
+            source: field.script as string,
```

Comment:
```suggestion
            source: field.script ?? '',
```
Might be worth defaulting instead of casting to ensure type safety here.

Reply:
I went with `source: field.script!` as its a smaller change than providing an empty string.